src/candid_rpc.rs

```diff
@@ -121,41 +143,44 @@ impl CandidRpcClient {
         })
     }
 
-    pub async fn eth_get_logs(&self, args: candid_types::GetLogsArgs) -> RpcResult<Vec<LogEntry>> {
+    pub async fn eth_get_logs(
+        &self,
+        args: candid_types::GetLogsArgs,
+    ) -> MultiRpcResult<Vec<LogEntry>> {
         let args: GetLogsParam = match args.try_into() {
             Ok(args) => args,
-            Err(err) => return Err(RpcError::from(err)),
+            Err(err) => return MultiRpcResult::Consistent(Err(RpcError::from(err))),
         };
-        wrap_result(self.client.eth_get_logs(args).await)
+        multi_result(self.client.eth_get_logs(args).await)
     }
 
-    pub async fn eth_get_block_by_number(&self, block: candid_types::BlockTag) -> RpcResult<Block> {
-        wrap_result(self.client.eth_get_block_by_number(block.into()).await)
+    pub async fn eth_get_block_by_number(
+        &self,
+        block: candid_types::BlockTag,
+    ) -> MultiRpcResult<Block> {
+        multi_result(self.client.eth_get_block_by_number(block.into()).await)
```

Comment:
understanding question: interesting, it was not clear to me until now that this calls the corresponding function `eth_get_block_by_number ` from your own branch of cketh. Do you plan on copying over the code into that repo so that ckETH can delete it when we will integeration with the EVM RPC canister or is that the part that's meant as a separate library?

Reply:
That's the plan, although this is going to be a huge lift due to the large number of types shared between the rest of ckETH and the Ethereum RPC logic. This corresponds to the "create shared library with ckETH" part of the roadmap, which was deferred until after completing the MVP.